### PR TITLE
Fix map_top_n to break ties by comparing keys

### PR DESCRIPTION
**Pull request by kgpai**
Presto Java internally breaks ties between equal non-null values in map_top_n by using the one with the larger key value.
(See : 
https://github.com/prestodb/presto/blob/0edb84cc8e2f825e257c558143edca1e154f8a72/presto-main/src/main/java/com/facebook/presto/operator/scalar/sql/MapSqlFunctions.java#L57 ). This change fixes it to be equivalent to presto behavior.

### CODE REVIEW

File: velox/functions/prestosql/tests/MapTopNTest.cpp [hunk 1]
```diff
@@ -66,5 +66,27 @@ TEST_F(MapTopNTest, basic) {
       "n must be greater than or equal to 0");
 }
 
+// Test to ensure we use keys to break ties if values are
+// equal.
+TEST_F(MapTopNTest, equalValues) {
```
Comment (kgpai) [inline velox/functions/prestosql/tests/MapTopNTest.cpp:71]: Working on adding more tests.

File: velox/functions/prestosql/MapTopN.h [hunk 2]
```diff
@@ -34,7 +34,13 @@ struct MapTopNFunction {
           CompareFlags::NullHandlingMode::kNullAsIndeterminate};
 
       if (l->second.has_value() && r->second.has_value()) {
-        return l->second.value().compare(r->second.value(), flags) > 0;
+        auto comp = l->second.value().compare(r->second.value(), flags);
+
+        if (FOLLY_UNLIKELY(comp == 0)) {
+          return l->first.compare(r->first, flags) > 0;
```
Comment (mbasmanova) [inline velox/functions/prestosql/MapTopN.h:40]: What happens if keys are not orderable?

Comment (kgpai) [reply velox/functions/prestosql/MapTopN.h:40]: Right now Presto throws an exception if you have keys that are not orderable : 
```
SELECT
    MAP_TOP_N(
        MAP(
            ARRAY[
                MAP(ARRAY[1], ARRAY[2]),
                MAP(ARRAY[3], ARRAY[2]),
                MAP(ARRAY[14], ARRAY[2])],
                ARRAY[1, 2, 3]),3 )
   ```
   
   I will validate and add test cases for that.

Comment (kgpai) [reply velox/functions/prestosql/MapTopN.h:40]: Changed signature to ensure keys are orderable.

File: velox/docs/functions/presto/map.rst [hunk 3]
```diff
@@ -96,7 +96,8 @@ Map Functions
 
 .. function:: map_top_n(map(K,V), n) -> map(K, V)
 
-    Truncates map items. Keeps only the top N elements by value.
+    Truncates map items. Keeps only the top N elements by value. Keys are used to break ties on values. Both Keys and Values should be orderable.
```
Comment (mbasmanova) [inline velox/docs/functions/presto/map.rst:99]: > Keys are used to break ties on values

Is this generally true or only for non-NULL values?

> Both Keys and Values

Do not capitalize Keys or Values: Both keys and values ...

File: velox/docs/functions/presto/map.rst [hunk 4]
```diff
@@ -96,7 +96,8 @@ Map Functions
 
 .. function:: map_top_n(map(K,V), n) -> map(K, V)
 
-    Truncates map items. Keeps only the top N elements by value.
+    Truncates map items. Keeps only the top N elements by value. Keys are used to break ties on non-NULL values. Both keys and values should be orderable.
```
Comment (kagamiori) [inline velox/docs/functions/presto/map.rst:99]: > Keys are used to break ties on non-NULL values.

Could you elaborate how keys are used to break ties? E.g., when there are multiple map entries with the same value, the entry with max/min key is taken.

**Review by kagamiori — APPROVED** [review]
LGTM. Please also make sure @mbasmanova's comments are addressed before landing.

**Pull request closed without merging**